index.js

```diff
@@ -4,9 +4,9 @@ import inherits from 'inherits'
  * Create a custom constructor which can be modified without affecting the
  * original class.
  *
- * @template {Function} Class
+ * @template {{ new(...args: any): any }} Class
  * @param {Class} Super
- * @returns {Class}
+ * @return {new(...args: ConstructorParameters<Class>) => InstanceType<Class>}
```

Comment:
TypeScript generics appear to strip constructors, this recreates the constructor through generic reflection